tests/starknet/executor.rs

```diff
@@ -0,0 +1,128 @@
+use std::{fs, thread};
+
+use anyhow::{anyhow, Error};
+use regex::Regex;
+
+use super::scarb::Compiler;
+
+#[allow(dead_code)]
+pub struct Executor {
+    accounts: Vec<String>,
+}
+
+#[allow(dead_code)]
+impl Executor {
+    pub fn new(num: usize) -> Result<Self, Error> {
```

Comment:
Why embed threading in the executor? It seems much easier to get an executor for a specific `id` and let client code decide if any number of them is necessary. Right now I don't really see the use case (please correct me if I'm wrong) for number of contracts deployed being anyhting that one :)

Reply:
Since the compilation is a blocking call and it goes into a separate thread, I started to think one step ahead. I wanted to make Executor reusable with simple API where the amount of contracts is going to be input and the rest of it will be done implicitly, including environment setup of followup files such as account.json, compilation by scarb, usage of starkli for declare and deploy, etc..
Additionally, before running this test case on live network, I plan to test it on Katana (hence, two tests, one for Katana and one for Sepolia). Since iterating multiple accounts compilation would take some time on a test case, with simple API in mind, this is the approach that I have decided on.
I agree with you, in the end we want to have only one contract that will be deployed. With this PR, I wanted to put everything on table, show what design questions came to my mind during the work and what we are dealing with. After the review, I could easily kick something out, if you thought that something was being overengineered.
To summarize, next steps that will be done, I will extract compilation outside of the executor and in the test case with Katana where we will deploy multiple accounts, compilation of the accounts is going to be called explicitly, directly inside the test case.

Reply:
@sergey-melnychuk Result of the changes is that `executor.rs` has been removed and that `coordinator.rs` is introduced. Coordinator's task is to store all paths and copy/update contract that is being worked on. I have removed support for multiple contract deployment and inside the test case explicit workflow can be checked. The one in `rpc.rs` that will be used to deploy accounts on sepolia is commented out, for now. I will uncomment it when we will test whole workflow on Katana.

Reply:
Please don't leave chunks of commented code in the pull-request, it is a hure code smell.

Extract it into separete issues/PRs if necessary.